ntbs-service/wwwroot/source/Components/AutocompleteSelect.ts

```diff
@@ -37,20 +42,31 @@ const AutocompleteSelect = Vue.extend({
             {
                 // Autocomplete does not automatically select blank value when deleting text rather than clicking on blank, so emit this manually
                 this.selectElement.value = "";
-                this.$emit("validate-input", event);
+                this.$emit("input-changed", event);
             }
         },
-        onConfirm: function (value: any) {
+        onConfirmValidate: function (value: any) {
             // this is called twice, value is undefined when user clicks away from input field which is when we want validation to happen
             if (!this.$props.validate || value != undefined) {
                 return;
             }
+            this.setSelectValue(this.autocompleteElement.value);
+        },
+        onConfirmFilter: function (value: any) {
+            // this is called twice, value is defined when user makes a selection which is when we want filtering to happen
+            if (!this.$props.filter) {
+                return;
+            }
+            if (value != undefined) {
+                this.setSelectValue(value);
+            }
+        },
+        setSelectValue: function (value: any) {
             const event = new Event('select-changed');
-            const autocompleteValue = this.autocompleteElement.value;
             // the actual select element does not have its value set at this point; the only way we seem to be able to find it is by searching for the matching autocomplete
             // value by name in all the options
             for (let i=0; i < this.selectElement.options.length; i++) {
-                if (autocompleteValue === this.selectElement[i].textContent) {
+                if (value === this.selectElement[i].textContent) {
                     this.selectElement.value = this.selectElement[i].value;
                     this.selectElement.dispatchEvent(event);

```

Comment:
I wonder if we should only call this when the select value has actually changed?

Reply:
Have added a check for a change in value